estate/models/estate_property.py

```diff
@@ -0,0 +1,100 @@
+        ('north', 'North'),
+        ('south', 'South'),
+        ('east', 'East'),
+        ('west', 'West'),
+    ], string='Garden Orientation')
+    total_area = fields.Integer(compute="_compute_total_area")
+    property_type_id = fields.Many2one('estate.property.type', string="Property Type")
+    salesperson_id = fields.Many2one('res.users', string="Salesperson", default=lambda self: self.env.user)
+    buyer_id = fields.Many2one('res.partner', string="Buyer", copy=False)
+    tag_ids = fields.Many2many('estate.property.tag', string="Tags")
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string="Offers")
+    best_price = fields.Float(compute="_compute_best_price")
+
+    # Functions
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends('offer_ids')
+    def _compute_best_price(self):
+        for record in self:
+            if record.offer_ids:
+                record.best_price = max(record.offer_ids.mapped('price'))
+            else:
+                record.best_price = 0
+
+    @api.onchange('garden')
+    def _onchange_partner(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = None
+            self.garden_orientation = None
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_if_property_new_or_cancelled(self):
+        for record in self:
+            if record.state not in ['new', 'cancelled']:
+                raise UserError(f"Can\'t delete property '{record.name}' if state is not New or Cancelled")
+
+    def action_sell_property(self):
+        for record in self:
+            if record.state == 'cancelled':
+                msg = "Cancelled properties cannot be sold."
+                raise UserError(msg)
+            record.state = 'sold'
+        return True
+
+    def action_cancel_property(self):
+        for record in self:
+            if record.state == 'sold':
+                msg = "Sold properties cannot be cancelled."
+                raise UserError(msg)
+            record.state = 'cancelled'
+        return True
+
+    # Constraints
+    _check_expected_price = models.Constraint('CHECK (expected_price > 0)', "A property expected price must be strictly positive")
+    _check_selling_price = models.Constraint('CHECK (selling_price >= 0)', "A property selling price must be positive")
+
+    @api.constrains('selling_price', 'expected_price')
+    def _check_selling_price_percentage(self):
+        for record in self:
+            if record.state == 'offer accepted' and float_compare(record.selling_price, 0.9 * record.expected_price, precision_digits=9) == -1:
+                msg = "The selling price cannot be lower than 90% of the expected price"
+                raise ValidationError(msg)
```

Comment:
I believe error with raised here in the runbot when you have the string directly. this because the Runbot expect a translation object as error msg. we should discuss this together 🫡 
```suggestion
                raise ValidationError(self,env._("The selling price cannot be lower than 90% of the expected price"))
```